src/utils/map-state.js

```diff
@@ -370,10 +362,10 @@ export default class MapState {
     let pitch = startPitch;
     if (deltaScaleY > 0) {
       // Gradually increase pitch
-      pitch = startPitch + deltaScaleY * (maxPitch - startPitch);
+      pitch = startPitch + Math.min(deltaScaleY, 1) * (maxPitch - startPitch);
```

Comment:
So there is no need to clamp `deltaScaleX` now that we removed the assert? Assume we apply modular arithmethic to keep it in range after we return? 

Will this allow us to keep dragging and rotating X when dragging the mouse outside of a window? If so, could be worth a comment.

Possibly, but not needed:
- move the parameter capping of `deltaScaleY` to one place, after `startZoom` on line 234?
- Use a helper function `clamp(.., -1, 1);`?

Reply:
This bug is especially easy to trigger when the map is not full screen, when dragging the mouse outside of the canvas. I don't see why we can't allow the bearing to continue rotating.

Reply:
>  I don't see why we can't allow the bearing to continue rotating.

Me neither, that is not what I was implying. I was just saying that it would be good to document in a comment that this is the intended behavior. The other comments related to `deltaScaleY`.